### PR TITLE
[Bugfix] Fixes R&D Machine Outputs If Flatpacked

### LINKED ISSUE #5086: (WEB REPORT BY: Anonymous REMOTE: ) Autolathe Items Not Spawning Correctly
>Revision 27295b13dc8f12c774720dedb2faee2b8b084e5f 

>General description of the issue
Autolathes created from flatpacks seem to be spawning any items they create at the Flatpack Fabricator rather than the autolathe itself.

>What you expected to happen
Tried to create a tile painter, followed by two crowbars. I expected the autolathe to take the required materials and after a short while complete the items before depositing them at my feet.

>What actually happened
Resources were taken from the autolathe, the lathe took the typical time to make the items, but no items appeared. After resturning to my workshop, I discovered the items laying in front of the Flatpack Fabricator. More items continue to appear there as other people attempt to use the lathe, even as I type this.

>Steps to reproduce if possible
The autolathe was placed between a single pane R-Window and a cyborg recharger I had also built. The autolathe was made from a flatpack.

### PR DESCRIPTION
**Pull request by Kurfursten**
@ComicIronic 

fixes  #5086

### CODE REVIEW

File: code/modules/research/mechanic/flatpack.dm [hunk 1]
```diff
@@ -80,6 +80,12 @@
 				instructions.update_icon()
 */
 				machine.forceMove(src.loc)
+				if(istype(machine,/obj/machinery/r_n_d)) //If it's something that outputs, like a fab, reset the output point
+					var/obj/machinery/r_n_d/R = machine
+					if(R.research_flags & HASOUTPUT)
+						if(istype(R.output,/obj/machinery/mineral/output))
+							del(R.output)
```
Comment (PJB3005) [inline code/modules/research/mechanic/flatpack.dm]: del

Comment (Kurfursten) [inline code/modules/research/mechanic/flatpack.dm]: Keeps consistency with the rest of comic's code, I took it right out of  his

**Comment by ComicIronic** [comment]
Augh, no, this is a patchwork fix.
The output system actually has to be changed completely to just use dirs, so it's completely dependant on loc instead of using output markers or anything like that.

**Comment by Kurfursten** [comment]
I'll close this if you want to fix it yourself, as long as you're willing to be timely about it. This one has been outstanding for a while.

File: code/modules/research/mechanic/flatpack.dm (hunk 1 above)
Comment (Iamgoofball) [inline code/modules/research/mechanic/flatpack.dm]: qdel or bust

**Comment by Kurfursten** [comment]
@ComicIronic ?

**Comment by Kurfursten** [comment]
Comic said over IRC that he's not fixing this, and he doesn't like my solution, so I guess the bug is here to stay.

**Pull request closed without merging**